bika/lims/browser/analysisrequest/analysisrequests.py

```diff
@@ -58,10 +58,16 @@ def __init__(self, context, request):
         self.contentFilter = {
             "sort_on": "created",
             "sort_order": "descending",
-            "path": {"query": "/", "depth": 2},
             "cancellation_state": "active",
         }
 
+        # Filter by Department
+        if self.context.bika_setup.getAllowDepartmentFiltering():
+            deps = self.request.get('filter_by_department_info', '')
+            dep_uids = deps.split(",")
+            dep_query = {"query": dep_uids, "operator": "or"}
+            self.contentFilter['getDepartmentUIDs'] = dep_query
```

Comment:
Do you mind to create a separate function somewhere else to deal with all this? Would be easier later to add changes without having to go through all code searching for this filter. I also think moving all this department filtering stuff to an add-on would be better. You could use an adapter here then to apply not only department filtering, but other additional filters we could eventually have in future without code changes in `senaite.core`

Reply:
@xispa , I just tried to make it more generic, but seems it would take some time to add a filtering Add-on. And since that enhancement about Filtering AR's in querying level is kind of important, I would suggest to have at least that change immediately... If you still think we should just follow the way creating new Add-On for filtering, I will start to work on it soon.